rafx-framework/src/resources/descriptor_sets/descriptor_set_allocator.rs

```diff
@@ -109,6 +110,24 @@ impl DescriptorSetAllocator {
         pool.insert(&self.device_context, write_set)
     }
 
+    pub fn create_descriptor_set_with_writer<'a, T: DescriptorSetWriter<'a>>(
+        &mut self,
+        descriptor_set_layout: &ResourceArc<DescriptorSetLayoutResource>,
+        args: T,
+    ) -> RafxResult<DescriptorSetArc> {
+        // Get or create the pool for the layout
```

Comment:
TODO: Can this code to acquire the pool be de-duplicated wrt `create_descriptor_set_with_writes`